### PR TITLE
Order detail feature

### PR DESCRIPTION
**Pull request by victuxbb**
* Refactor order domain and his factory to allow creation using optional named params
* Add readonly flag in new order state to handle edition feature
* Add conditionals on react to show print button instead of save button, also remove delete action and make quantity input readonly
* Add epic flow to print the current order
* Fix orders fixtures to use unique id
* Add new order reducer test

### CODE REVIEW

File: src/openMarket/infrastructure/order/LocalStorageOrderRepository.js [hunk 1]
```diff
@@ -13,7 +13,15 @@ export default class LocalStorageOrderRepository extends OrderRepository {
   }
 
   findById({id}) {
-    return null;
+    return RxLocalStorage.loadLocalStorage({ localStorageKey: this._localStorageKey })
```
Comment (Arkioner) [inline src/openMarket/infrastructure/order/LocalStorageOrderRepository.js:16]: So this should be injected or this class renamed to RxLocalStorageOrderRepository

File: src/openMarket/domain/order/Order.js [hunk 2]
```diff
@@ -5,19 +5,20 @@ import moment from "moment";
 export default class Order {
 
   /**
-   * @param {IdGenerator} idGenerator
+   * @param {String} id
    * @param {Array.<Line>} lines
+   * @param {date} date
    */
-  constructor({ idGenerator, lines }) {
+  constructor({ id, lines, date=moment().format("DD/MM/YYYY HH:mm:ss") } = {}) {
```
Comment (Arkioner) [inline src/openMarket/domain/order/Order.js:12]: I don't understand the part of date :( could you explain in less than 140 characters. Please not brakes :D

File: src/openMarket/infrastructure/order/OrderFactoryImpl.js [hunk 3]
```diff
@@ -7,11 +7,11 @@ export default class OrderFactoryImpl extends OrderFactory {
     this._identity = identity;
   }
 
-  createWith({lines}) {
-    return new Order(
-      {
-        idGenerator: this._identity,
-        lines: lines
+  createWith({id = this._identity.generate(), lines, date} = {}) {
```
Comment (Arkioner) [inline src/openMarket/infrastructure/order/OrderFactoryImpl.js:10]: id is given and if not the default value is the identity generate? Just to understand

File: src/openMarket/user_interface/order/list_orders/Container.js [hunk 4]
```diff
@@ -29,10 +30,21 @@ class Container extends Component {
         Header: 'Total',
         accessor: 'total',
         Cell: (data) => data.value +" "+this.moneySymbol
+      },
+      {
+        Header: 'Actions',
+        accessor: 'actions',
+        Cell: this.renderDetailOrder
```
Comment (Arkioner) [inline src/openMarket/user_interface/order/list_orders/Container.js:37]: That looks pretty cool 🥇

File: src/openMarket/infrastructure/order/LocalStorageOrderRepository.js (hunk 1 above)
Comment (victuxbb) [reply src/openMarket/infrastructure/order/LocalStorageOrderRepository.js:16]: you are right, this should be injected, I can't remember why the first implementation of RxLocalStorage is static, anyway I need to apply a big refactor in all repos based on RxLocalStorage
:+1:

File: src/openMarket/domain/order/Order.js (hunk 2 above)
Comment (victuxbb) [reply src/openMarket/domain/order/Order.js:12]: Yes, it's related with named parameters and default parameters in ES6.
If date key is not provided in constructor, the default value for this key is getting the current date using moment library.
Perhaps you wanna take a look at this http://jimrottinger.github.io/named-arguments-using-es6-destructuring/

File: src/openMarket/infrastructure/order/OrderFactoryImpl.js (hunk 3 above)
Comment (victuxbb) [reply src/openMarket/infrastructure/order/OrderFactoryImpl.js:10]: Yes, is like the previous case in domain Order constructor :+1:

**Pull request merged**